src/components/Firestore/DocumentEditor/index.tsx

```diff
@@ -299,6 +303,14 @@ const FieldEditor: React.FC<{
         </div>
       </div>
     );
+  } else if (isJson) {
```

Comment:
What do you think of separating json from the "primitive" fields? Then we don't need to have an isJson prop on these components

Reply:
Right now the fieldEditor uses a hook to get a field from the store, and those fields can be map/array/primitive; with corresponding if-blocks in this component. What if we add a fourth field-type to the store (json) and add a fourth if-else block in this component that renders the JsonEditor?

Reply:
That was my intuition but wasn't sure if it passed the sniff test of your design. Yes that sounds good to me!